repository/kafka/src/main/java/com/expediagroup/streamplatform/streamregistry/repository/kafka/KafkaConfiguration.java

```diff
@@ -37,40 +37,43 @@ EventCorrelator eventCorrelator() {
 
   @Bean
   EventSender eventSender(
-      @Value("${repository.kafka.bootstrapServers}") String bootstrapServers,
-      @Value("${repository.kafka.topic:_streamregistry}") String topic,
-      @Value("${repository.kafka.schemaRegistryUrl}") String schemaRegistryUrl,
-      EventCorrelator eventCorrelator
+    @Value("${repository.kafka.bootstrapServers}") String bootstrapServers,
+    @Value("${repository.kafka.topic:_streamregistry}") String topic,
+    @Value("${repository.kafka.schemaRegistryUrl}") String schemaRegistryUrl,
+    EventCorrelator eventCorrelator
   ) {
     KafkaEventSender.Config config = KafkaEventSender.Config.builder()
-        .bootstrapServers(bootstrapServers)
-        .topic(topic)
-        .schemaRegistryUrl(schemaRegistryUrl)
-        .build();
+      .bootstrapServers(bootstrapServers)
+      .topic(topic)
+      .schemaRegistryUrl(schemaRegistryUrl)
+      .build();
     return new KafkaEventSender(config, eventCorrelator);
   }
 
   @Bean
   EventReceiver eventReceiver(
-      @Value("${repository.kafka.bootstrapServers}") String bootstrapServers,
-      @Value("${repository.kafka.topic:_streamregistry}") String topic,
-      @Value("${repository.kafka.groupId:stream-registry}") String groupId,
-      @Value("${repository.kafka.schemaRegistryUrl}") String schemaRegistryUrl,
-      EventCorrelator eventCorrelator
+    @Value("${repository.kafka.bootstrapServers}") String bootstrapServers,
+    @Value("${repository.kafka.topic:_streamregistry}") String topic,
+    @Value("${repository.kafka.groupId:stream-registry}") String groupId,
+    @Value("${repository.kafka.schemaRegistryUrl}") String schemaRegistryUrl,
+    EventCorrelator eventCorrelator
   ) {
     KafkaEventReceiver.Config receiverConfig = KafkaEventReceiver.Config.builder()
-        .bootstrapServers(bootstrapServers)
-        .topic(topic)
-        .groupId(groupId)
-        .schemaRegistryUrl(schemaRegistryUrl)
-        .build();
+      .bootstrapServers(bootstrapServers)
+      .topic(topic)
+      .groupId(groupId)
+      .schemaRegistryUrl(schemaRegistryUrl)
+      .build();
     return new KafkaEventReceiver(receiverConfig, eventCorrelator);
   }
 
   @Bean
   EntityView entityView(EventReceiver eventReceiver) {
     EntityView entityView = new DefaultEntityView(eventReceiver);
-    entityView.load().join();
+    PurgingEntityViewListener entityViewListener = new PurgingEntityViewListener(entityView);
```

Comment:
Two way dependency `PurgingEntityViewListener` <-> `EntityView` but I don't think we can avoid that.

Reply:
Yeah, agreed this is a bit pants. I originally tried to hide this by putting the purging in the constructor of `PurgingEntityViewListener` but I think that looked worse.